core/src/main/scala/akka/persistence/jdbc/state/scaladsl/JdbcDurableStateStore.scala

```diff
@@ -60,8 +60,8 @@ class JdbcDurableStateStore[A](
         .flatMap(r => db.run(queries._updateDurableState(r)))
         .map { rowsAffected =>
           if (rowsAffected == 0)
-            throw new Exception(
-              s"Incorrect sequence number $seqNr provided: $seqNr has to be 1 more than the value existing in the database for this $persistenceId")
+            throw new IllegalArgumentException(
+              s"Incorrect sequence number [$seqNr] provided: It has to be 1 more than the value existing in the database for persistenceId $persistenceId")
```

Comment:
I think this message is targeting the final user, but the final user is not really in control of which seqNr is being passed. So they can't control that it needs to be 1 more than the value in the DB.

We will run into this situation if we have a cluster split brain or a user using the `DurableStateBehavior` without a cluster and not taking care of duplicated instances. 

Instead, we can be more explicit about it, something like: 

> Incorrect sequence number [$seqNr] provided. Another instance of that same durable state actor is probably running in your system.

Reply:
Suggest adding both the messages like 
```scala
throw new IllegalArgumentException(              
  s"Incorrect sequence number [$seqNr] provided: It has to be 1 more than the value existing in the database for persistenceId $persistenceId. Maybe another instance of that same durable state actor is running in your system")
```

@patriknw @octonato ?